ecl/hql/hqlopt.cpp

```diff
@@ -3325,14 +3325,14 @@ IHqlExpression * CTreeOptimizer::doCreateTransformed(IHqlExpression * transforme
             case no_subsort:
                 if (transformedCountProject)
                     break;
-                if (increasesRowSize(transformed))
+                if ((options & HOOminimizeNetworkAndMemory) && increasesRowSize(transformed))
```

Comment:
If the new row size is unknown it should default to assuming the size is increased.  I didn't come up with a good way of naming the function.